src/ripple/app/paths/PathRequest.cpp

```diff
@@ -408,6 +446,110 @@ void PathRequest::resetLevel (int l)
         iLastLevel = l;
 }
 
+bool
+PathRequest::findPaths (
+    RippleLineCache::ref cache,
+        FindPaths& fp, Issue const& issue,
+            Json::Value& jvArray)
+{
+    // TODO MPORTILLA Is it necessary to cache this iusse's STPathSet?
+    // Is the unbound map growth a potential problem?
+    STPath fullLiquidityPath;
```

Comment:
The map growth is not unbounded. There will only be one entry for each currency the source account holds. And the caching is absolutely critical. Without it, if an account does issuer-specific pathfinding, we'll repeat precisely the same pathfinding effort for every issuer.